src/plugins/template/src/plugin.cpp

```diff
@@ -25,6 +29,32 @@ namespace {
 static constexpr const char* wait_executor_name = "TemplateWaitExecutor";
 static constexpr const char* stream_executor_name = "TemplateStreamsExecutor";
 static constexpr const char* template_exclusive_executor = "TemplateExecutor";
+
+uint64_t get_blob_data_size(std::istream& model) {
+    uint64_t size = 0;
+    model.read(reinterpret_cast<char*>(&size), sizeof(size));
+    return size;
+}
```

Comment:
there is no guarantee that stream is on the start. 
I suggest to rewind it to start, get value and rewind back.
smth like:
```
auto pos = model.tellg();
model.seekg(0);
uint64_t size = 0;
model >> size;
model.seekg(pos, std::ios_base::cur);
return size;
```

Reply:
Is expected that stream is not on start are this helper is used twice for model size and weights size which are stored in different places.
This is just extract of current implementation as is used in more then one place.